etl/processor/processor.go

```diff
@@ -0,0 +1,100 @@
+package processor
+
+import (
+	"context"
+
+	"github.com/insolar/block-explorer/etl/interfaces"
+	"github.com/insolar/block-explorer/etl/models"
+	"github.com/insolar/block-explorer/etl/types"
+)
+
+type Processor struct {
+	JDC     <-chan types.JetDrop
+	TaskC   chan Task
+	Workers int
+	Storage interfaces.Storage
+}
+
+func NewProcessor(jb interfaces.Transformer, storage interfaces.Storage, workers int) *Processor {
+	if workers < 1 {
+		workers = 1
+	}
+	return &Processor{
+		JDC:     jb.GetJetDropsChannel(),
+		Workers: workers,
+		Storage: storage,
+	}
+
+}
+
+func (p *Processor) Start(ctx context.Context) error {
+	p.TaskC = make(chan Task)
+	for i := 0; i < p.Workers; i++ {
+		go func() {
+			for {
+				t, ok := <-p.TaskC
+				if !ok {
+					return
+				}
+				p.Process(t.JD)
+			}
+		}()
+	}
+
+	go func() {
+		for {
+			jd, ok := <-p.JDC
+			if !ok {
+				close(p.TaskC)
```

Comment:
You can close channel `p.TaskC` twice: if `Transformer` close `p.JDC` and `Processor.Stop` will be called. It will lead to panic.